### PR TITLE
Add option to check H2O version with h2o_context()

### PR DESCRIPTION
**Pull request by navdeep-G**
* This PR makes `strict_version_check` a parameter for `h2o_context`.

### CODE REVIEW

File: r/rsparkling/R/h2o_context.R [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+#' @param strict_version_check (Optional) Setting this to FALSE does not cross check version of H2O and attempts to connect.
 #' @export
-h2o_context <- function(x) {
+h2o_context <- function(x, strict_version_check = TRUE) {
   UseMethod("h2o_context")
 }
 
 #' @export
-h2o_context.spark_connection <- function(x) {
+h2o_context.spark_connection <- function(x, strict_version_check = TRUE) {
```
Comment (jangorecki) [inline r/rsparkling/R/h2o_context.R]: argument `strict_version_check` was not passed in a function few lines below this one.

Comment (navdeep-G) [inline r/rsparkling/R/h2o_context.R]: Check my latest commit

**Comment by jangorecki** [comment]
we forget about the need to render Rd files and include them in git commit, thus we now get
```
* checking S3 generic/method consistency ... WARNING
h2o_context:
  function(x, strict_version_check)
h2o_context.spark_jobj:
  function(x)

* checking for code/documentation mismatches ... WARNING
Codoc mismatches from documentation object 'h2o_context':
h2o_context
  Code: function(x, strict_version_check = TRUE)
  Docs: function(x)
  Argument names in code not in docs:
    strict_version_check
```

**Comment by ledell** [comment]
@navdeep-G @jangorecki Maybe we need to add an `R CMD check` to the testing framework for rsparkling?  That would have caught this, and we do that for the h2o R package.

**Pull request merged**